osbs/api.py

```diff
@@ -451,35 +484,33 @@ def _do_create_prod_build(self, git_uri, git_ref,
                               is_auto=False,
                               koji_parent_build=None,
                               isolated=None,
+                              flatpak=False,
+                              module=None,
+                              module_compose_id=None,
                               **kwargs):
+
         repo_info = utils.get_repo_info(git_uri, git_ref, git_branch=git_branch)
-        df_parser = repo_info.dockerfile_parser
         build_request = self.get_build_request(inner_template=inner_template,
                                                outer_template=outer_template,
                                                customize_conf=customize_conf)
-        labels = utils.Labels(df_parser.labels)
 
-        required_missing = False
-        req_labels = {}
-        # version label isn't used here, but is required label in Dockerfile
-        # and is used and required for atomic reactor
-        # if we don't catch error here, it will fail in atomic reactor later
-        for label in [utils.Labels.LABEL_TYPE_NAME,
-                      utils.Labels.LABEL_TYPE_COMPONENT,
-                      utils.Labels.LABEL_TYPE_VERSION]:
-            try:
-                _, req_labels[label] = labels.get_name_and_value(label)
-            except KeyError:
-                required_missing = True
-                logger.error("required label missing from Dockerfile : %s",
-                             labels.get_name(label))
+        if flatpak:
+            if module is None:
+                raise ValueError("Flatpak build missing required parameter 'module'")
+            if isolated:
+                # Flatpak builds from a particular stream autogenerate the release
+                # as <module_version>.<n>; it doesn't make sense to make a fix
+                # from specific one of these autogenerated version. What an isolated
+                # fix for module requires will have to be determined from experience.
+                raise ValueError("Flatpak build cannot be isolated")
```

Comment:
Do we want test coverage for either of these two failure paths?

Reply:
Add a a test case that exercises the command-line => build args pathj and tests these two cases and also causes the conf.py code to be called.